logic/src/androidMain/kotlin/com/wire/kalium/logic/CoreLogic.kt

```diff
@@ -23,16 +23,22 @@ actual class CoreLogic(
     private val applicationContext: Context,
     clientLabel: String,
     rootProteusDirectoryPath: String,
-) : CoreLogicCommon(clientLabel, rootProteusDirectoryPath) {
+    kaliumLoggerConfig: KaliumLogger.Config
+) : CoreLogicCommon(clientLabel, rootProteusDirectoryPath, kaliumLoggerConfig) {
+
     override fun getAuthenticationScope(): AuthenticationScope =
-        AuthenticationScope(clientLabel = clientLabel, applicationContext = applicationContext)
+        AuthenticationScope(
+            clientLabel = clientLabel,
+            applicationContext = applicationContext,
+            kaliumLogger = kaliumLogger
```

Comment:
Why do we need to pass around the instance of `kaliumLogger`? Can't we get it as a Singleton instance created on the WireApplication level?

Reply:
New module-wide `[Module]Logger` object was added to hold an instance instead of passing it down to where it needed.

You can check: `CoreLogger` specifically, for more info on how it was implemented, and also `ListenToEventsUseCase` on how it is now being used.